src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/DefaultOpenDistroSecurityKeyStore.java

```diff
@@ -522,14 +531,43 @@ public SSLEngine createServerTransportSSLEngine() throws SSLException {
         final SSLEngine engine = transportServerSslContext.newEngine(PooledByteBufAllocator.DEFAULT);
         engine.setEnabledProtocols(getEnabledSSLProtocols(this.sslTransportServerProvider, false));
         return engine;
+    }
 
+    /**
+     * Cert Issuer, Subject and Principal DN must be the same
+     * @param newCerts Array of X509Certificates which will replace our current cert
+     * @throws Exception if certificate is invalid
+     */
+    private void validateNewTransportCerts(final X509Certificate[] newCerts) throws Exception {
+
+        // First time we init certs ignore validity check
+        if (transportCerts == null) {
+            return;
+        }
+
+        // Check if new X509 certs have been issued by the same Issuer DN - Subject DN - Principal
+
+        // Create a sorted array of string with format IssuerDN/SubjectDN/Principal
+        final List<String> currentCertDNList = Arrays.stream(transportCerts)
+            .map(c -> c.getIssuerDN()+"/"+c.getSubjectDN()+"/"+c.getIssuerX500Principal().getName())
+            .sorted()
+            .collect(Collectors.toList());
+        final List<String> newCertDNList = Arrays.stream(newCerts)
+            .map(c -> c.getIssuerDN()+"/"+c.getSubjectDN()+"/"+c.getIssuerX500Principal().getName())
+            .sorted()
+            .collect(Collectors.toList());
+
+        // Both arrays must be equal
+        if (!currentCertDNList.equals(newCertDNList)) {
+            throw new Exception("DNs of new cert do not match");
+        }
```

Comment:
Should we also validate expiry? (expiry of a new cert should be later than old cert)

Reply:
Thanks for the suggestion. Have added a check for expiry as well.